lxd/images.go

```diff
@@ -984,8 +1061,66 @@ func doDeleteImage(d *Daemon, fingerprint string) error {
 func imageDelete(d *Daemon, r *http.Request) Response {
 	fingerprint := mux.Vars(r)["fingerprint"]
 
+	deleteFromAllPools := func() error {
+		// Use the fingerprint we received in a LIKE query and use the full
+		// fingerprint we receive from the database in all further queries.
+		_, imgInfo, err := dbImageGet(d.db, fingerprint, false, false)
+		if err != nil {
+			return err
+		}
+
+		poolIDs, err := dbImageGetPools(d.db, imgInfo.Fingerprint)
+		if err != nil {
+			return err
+		}
+
+		if len(poolIDs) == 0 {
+			// Use the fingerprint we received in a LIKE query and use the full
+			// fingerprint we receive from the database in all further queries.
+			id, imgInfo, err := dbImageGet(d.db, fingerprint, false, false)
```

Comment:
Save yourself a DB query and just store the id from the exact same dbImageGet 15 lines prior :)